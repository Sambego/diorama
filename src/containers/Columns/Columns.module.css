.columns {
  display: flex;
  flex-direction: column;
  width: 100vw;
  height: 100vh;
  min-height: 400px;
}

.column {
  flex: 1;
  display: flex;
  align-items: center;
  justify-content: center;
  flex-direction: column;
  width: 100%;
  height: 100%;
  text-align: center;
}

@media (min-width: 600px) {
  .columns {
    flex-direction: row;
  }
}

[class~='diorama-presenter-preview'] {
  .columns {
    flex-direction: row;
    min-height: initial;
  }
}
