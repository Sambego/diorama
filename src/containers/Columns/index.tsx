import React, { cloneElement, Children } from "react";
import PropTypes from "prop-types";

import styles from "./Columns.module.css";

type ColumnsProps = {
	className?: string;
	children: React.ReactNode;
};

function Columns({ children, className }: ColumnsProps) {
	const renderColumns = () =>
		Children.map(children, (child, index) => {
			// TODO: Not sure of what this style is supposed to do with children
			if (React.isValidElement(child)) {
				return cloneElement(child, {
					className: `${styles.column} diorama-column ${className} `,
					// eslint-disable-next-line react/no-array-index-key
					key: index,
					// @ts-ignore
					style: { maxWidth: `${100 / React.Children.count(children)}%` },
				});
			}
			return null;
		});

	return (
		<div className={`${styles.columns} diorama-columns`}>{renderColumns()}</div>
	);
}

Columns.propTypes = {
	children: PropTypes.node.isRequired,
	className: PropTypes.string,
};

Columns.defaultProps = {
	className: "",
};

export default Columns;
