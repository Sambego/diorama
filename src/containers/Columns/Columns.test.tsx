import React from "react";
import renderer from "react-test-renderer";
import Columns from ".";

it("renders correctly", () => {
	const tree = renderer
		.create(
			<Columns>
				<div>Column 1</div>
				<div>Column 2</div>
			</Columns>
		)
		.toJSON();
	expect(tree).toMatchSnapshot();
});
