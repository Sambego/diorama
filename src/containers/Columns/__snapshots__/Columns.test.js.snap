// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="columns diorama-columns"
>
  <div
    className="column diorama-column  "
    style={
      Object {
        "maxWidth": "50%",
      }
    }
  >
    Column 1
  </div>
  <div
    className="column diorama-column  "
    style={
      Object {
        "maxWidth": "50%",
      }
    }
  >
    Column 2
  </div>
</div>
`;
